plugins/esi/esi.cc

```diff
@@ -139,6 +154,7 @@ struct ContData {
       input_type(DATA_TYPE_RAW_ESI),
       packed_node_list(""),
       gzipped_data(""),
+      dbg_ctl{createDebugTag(DEBUG_TAG, contptr)},
```

Comment:
Won't this insert a lot of tags into the registry? How often is `ContData` instantiated?

Reply:
If you create multiple DbgCtl instances for the same tag, they will all point to the same entry in the registry map.  In any case, I think each instance of this continuation class will have a unique debug tag.  So there's a net increase, per class instance, of 8 bytes for the pointer in the DbgCtl, plus the overhead of an additional std::map entry.

Reply:
It looks like a new ContData instance is created per transaction that this plugin handles.  Since the tag is created using the address of the `contptr` instance, I feel like there is a good chance that there will be a lot of additional tags due to this.

Reply:
But my changes don't increase the number of instances of ContData.  I see the issue, but what can I do about it?

Reply:
It seems like these tags are generated to add correlation context to the log statements and not as a way to filter them with the tagging mechanism.  One idea would be to move this context from the tag name into the log message and have a fixed number of dbgctl tags.

Reply:
But does that really belong in *this* PR?  The change you suggest would reduce memory use in the existing code.  It can't clearly be claimed it's addressing an issue that I created.

Reply:
OK, valid point.

Reply:
Fixed